ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/Claims.scala

```diff
@@ -85,41 +86,63 @@ final case class Claims(
     participantId: Option[String] = None,
     expiration: Option[Instant] = None,
 ) {
-  def validForLedger(id: String): Boolean =
-    ledgerId.forall(_ == id)
+  protected val logger: Logger = LoggerFactory.getLogger(AuthServiceJWT.getClass)
+
+  def validForLedger(id: String): Boolean = {
+    logOnFailure(
+      ledgerId.forall(_ == id),
+      "ledgerId",
+      s"${ledgerId.getOrElse("<EMPTY>")} does not match expected value $id")
+  }
 
   def validForParticipant(id: String): Boolean =
-    participantId.forall(_ == id)
+    logOnFailure(
+      participantId.forall(_ == id),
+      "participantId",
+      s"${participantId.getOrElse("<EMPTY>")} does not match expected value $id")
 
-  /** Returns false if the expiration timestamp exists and is greather than or equal to the current time */
+  /** Returns false if the expiration timestamp exists and is greater than or equal to the current time */
   def notExpired(now: Instant): Boolean =
-    expiration.forall(now.isBefore)
+    logOnFailure(
+      expiration.forall(now.isBefore),
+      "expiration timestamp",
+      s"$expiration is not greater than or equal to the current time $now")
 
   /** Returns true if the set of claims authorizes the user to use admin services, unless the claims expired */
   def isAdmin: Boolean =
-    claims.contains(ClaimAdmin)
+    logOnFailure(claims.contains(ClaimAdmin), "admin", "not authorized to use services")
 
   /** Returns true if the set of claims authorizes the user to use public services, unless the claims expired */
   def isPublic: Boolean =
-    claims.contains(ClaimPublic)
+    logOnFailure(claims.contains(ClaimPublic), "public", "not authorized to use services")
 
   /** Returns true if the set of claims authorizes the user to act as the given party, unless the claims expired */
   def canActAs(party: String): Boolean = {
-    claims.exists {
+    logOnFailure(claims.exists {
```

Comment:
Note that `canActAs` and `canReadAs` are ran in a loop by the `Authorizer`, so if you have a request that asks to read on behalf of `Alice` and `Bob` and a token that allows that, you'll still see a log line stating that that you're not able to read as `Alice` and as `Bob`. Could cause some head-scratching.